### PR TITLE
fix: Fix L0_request_cancellation and L0_backend_output_detail byte size checks

### PR DESCRIPTION
**Pull request by rmccorm4**
#### What does the PR do?

Recent byte size validation changes caught issues with the input metadata being provided by these tests.

This PR fixes this:
1. Fixed input0_data vector to match the expected input shape L0_request_cancellation
2. Fixed byte size passed by both tests to [AppendInputData](https://github.com/triton-inference-server/core/blob/bbcd7816997046821f9d1a22e418acb84ca5364b/include/triton/core/tritonserver.h#L1298)

#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [X] Verified copyright is correct on all changed files.
- [x] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [X] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [x] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [x] test

#### Related PRs:
None

#### Where should the reviewer start?
N/A

#### Test plan:
Pre-existing tests

#### Caveats: 
None

#### Background
None

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
None

### CODE REVIEW

**Review by jbkyang-nvi — DISMISSED** [review]
thanks for the change and adhering to the template 🙏

File: src/test/request_cancellation_test.cc [hunk 1]
```diff
@@ -175,7 +177,7 @@ class RequestCancellationTest : public ::testing::Test {
 };
 
 TRITONSERVER_Server* RequestCancellationTest::server_ = nullptr;
-std::vector<int32_t> RequestCancellationTest::input0_data_(16, 1);
+std::vector<int32_t> RequestCancellationTest::input0_data_(1000, 1);
```
Comment (jbkyang-nvi) [inline src/test/request_cancellation_test.cc:180]: Q: What's the point of changing it here?

Comment (rmccorm4) [reply src/test/request_cancellation_test.cc:180]: It has to match the `    std::vector<int64_t> input0_shape({1, 1000});` , which is also the shape specified in the `config.pbtxt` of the model being used for testing

**Review by kthui — APPROVED** [review]
Thanks for fixing request cancellation!

**Review by kthui — APPROVED** [review]

**Pull request merged**